csi/server/plugin/opensds/node.go

```diff
@@ -242,49 +263,66 @@ func (p *Plugin) NodeStageVolume(
 
 	// Check if it is: "Volume published but is incompatible"
 	mnt := req.VolumeCapability.GetMount()
-	mountFlags := mnt.MountFlags
-	_, err = exec.Command("findmnt", device, mountpoint).CombinedOutput()
-	glog.V(5).Infof("findmnt err: %v \n", err)
+	block := req.VolumeCapability.GetBlock()
+	glog.V(5).Infof("VolumeCapability Mount=%+v, Block=%+v\n", mnt, block)
 
-	if nil == err {
-		if len(mountFlags) > 0 {
-			_, err := exec.Command("findmnt", "-o", strings.Join(mountFlags, ","), device, mountpoint).CombinedOutput()
-			if nil != err {
-				return nil, status.Error(codes.Aborted, "Volume published but is incompatible")
-			}
-		}
-
-		return &csi.NodeStageVolumeResponse{}, nil
+	if (nil != mnt) && (nil != block) {
+		return nil, status.Error(codes.InvalidArgument, "volumeMode cannot be both Block and Filesystem")
 	}
 
-	// Format
-	curFSType := connector.GetFSType(attachment.Mountpoint)
-	hopeFSType := DefFSType
-	if "" != mnt.FsType {
-		hopeFSType = mnt.FsType
+	if nil == vol.Metadata {
+		vol.Metadata = make(map[string]string)
 	}
 
-	if "" == curFSType {
-		_, err := exec.Command("mkfs", "-t", hopeFSType, "-F", device).CombinedOutput()
-		if err != nil {
-			return nil, status.Error(codes.Aborted, fmt.Sprintf("failed to mkfs: %v", err.Error()))
+	var mountFlags []string
+	if nil == block {
+		vol.Metadata[KCSIVolumeMode] = "Filesystem"
+		mountFlags = mnt.MountFlags
+		_, err = exec.Command("findmnt", device, mountpoint).CombinedOutput()
+		glog.V(5).Infof("findmnt err: %v \n", err)
```

Comment:
We are not sure if there is an error or not to log here.  If this is for debugging, I think you should add "device" and "mountpoint" in the log msg as well.

Reply:
Returning err here is not an error, it just means that the volume has been published.